migration-notes.md

```diff
@@ -0,0 +1,260 @@
+
+With rainbowkit using wagmi under the hood, the possiblities are pretty much endless, so here is a straightforward example trying to read data from a contract.
+
+```diff
+- import { useCelo } from '@celo/react-celo';
++ import { useAccount } from 'wagmi';
++ import { accountsABI } from '@celo/abis/types/wagmi';
++ import useCeloRegistryAddress from './use-celo-registry-address';
+
++ function useAccountSummary() {
++     const { isConnected, address } = useAccount();
++     const { data: accountSummary } = useContractRead({
++         address: useCeloRegistryAddress('Accounts'),
++         abi: accountsABI,
++         functionName: 'getAccountSummary',
++     });
++
++     return accountSummary
++ }
+
+function App() {
+-  const { kit, address } = useCelo();
+-
+-  async function getAccountSummary() {
+-    const accounts = await kit.contracts.getAccounts();
+-    await accounts.getAccountSummary(address);
+-  }
++  const accountSummary = useAccountSummary();
+
+  return (
+    ...
+  )
+}
+```
+
+### Send a transaction
+
+```ts
+// react-celo
+import { useCelo } from '@celo/react-celo';
+
+function App() {
+  const { performActions } = useCelo();
+
+  async function transfer() {
+    await performActions(async (kit) => {
+      const cUSD = await kit.contracts.getStableToken();
+      await cUSD.transfer('0x...', 10000).sendAndWaitForReceipt();
+    });
+  }
+
+  return <button onClick={transfer}>Transfer</button>;
+}
+```
+
+Rainbowkit doesn't support a similar function as `performActions` out of the box, but you may achieve a similar result using `react-modal` and a simple `useState`.
+
+```ts
+// rainbowkit
+import { usePrepareContractWrite, useContractWrite } from 'wagmi';
+import { StableTokenABI } from '@celo/abis/types/wagmi';
+import useCeloRegistryAddress from './use-celo-registry-address';
+// optional
+import MyCustomModal from './modal';
+
+function App() {
+  // optional modal state
+  const [isOpen, setIsOpen] = useState(false);
+
+  const { config } = usePrepareContractWrite({
+    address: useCeloRegistryAddress('StableToken'),
+    abi: StableTokenABI,
+    functionName: 'transfer',
+    args: ['0x...', 10000],
+    onSettle: () => setIsOpen(false),
+  });
+  const { write } = useContractWrite(config);
+  const transfer = useCallback(() => {
+    setIsOpen(true);
+    write();
+  }, [write]);
+
+  return;
+  <>
+    <button onClick={transfer}>Transfer</button>
+    {isOpen && <MyCustomModal />}
+  </>;
+}
+```
+
+### Fee currency
+
+While react-celo provides a `feeCurrency` variable and an `updateFeeCurrency` helper method, this isn't the case for rainbowkit. However, rainbowkit also supports `feeCurrency` out of the box thanks to its Celo-specific block and transactions formatters. You can find an advanced example in the [`rainbowkit-celo` package right here](https://rainbowkit-with-celo.vercel.app/fee-currency).
```

Comment:
thinking out loud: i wonder if we should add something to rainbowkit-celo for this. or make the example i made more rainbowkit specific (now its more viem specific)

Reply:
I'll do a better rainbowkit-celo example in a different PR in the rainbowkit-celo repo then, I think it's more useful this way

Reply:
edit from our in video meetings: kinda hit a wall, and the next steps in the discussion with the wagmi team: 
- https://github.com/wagmi-dev/wagmi/discussions/2817